Sources/PayPalWebPayments/PayPalWebCheckoutClient.swift

```diff
@@ -104,6 +104,10 @@ public class PayPalWebCheckoutClient: NSObject {
     /// - Returns: PayPalVaultResult
     /// - Throws: PayPalSDK error if vaulting could not complete successfully
     public func vault(url: URL) {
+        if let setupToken = getQueryStringParameter(url: url.absoluteString, param: "approval_session_id") {
+            analyticsService = AnalyticsService(coreConfig: config, setupToken: setupToken)
+        }
```

Comment:
Also slightly unrelated to analytics (but revealed by analytics 😆) - should we boot a merchant from the flow if their URL doesn't contain this value? Currently merchants can pass us any URL and we'll open it in an ASWebSession.

Reply:
+1 for being precautious. The only risk here would be if the format of the URL we get from PPaaS changed one day to something we don't expect, it could break existing integrations.

Reply:
Agree, seems like a pretty big risk vector. You could pass in `stealyourcreditcard.com` and we'd be like "sounds good!" and open it right up.

Reply:
Y'all know what we would need to allowlist so we can patch this?

Reply:
Luckily this hasn't been released yet on iOS, so we should be able to resolve this. We also should figure out when we want to release since our unreleased section is real long and we probably should be doing smaller releases along the way.

Reply:
Yeah on second thought maybe restricting the TLD to `paypal.com` at least? That way we know for certain it's going to us over TLS.

Reply:
Captured in DTPPCPSDK-1986.